analysis/preview.py

```diff
@@ -223,19 +239,25 @@ def main():
     args = parse_args()
     jdoc = get_configuration(args.input_file)
 
-    zsps, drones = parse_positions(jdoc)
+    zsps, drones, step = parse_positions(jdoc)
 
     ax = setup_3dplot()
     build_zsp_space(ax, zsps)
 
     trajectories = []
+    i=0
     for drone in drones:
-        trajectories.append(build_drone_trajectory(drone))
+        trajectories.append(build_drone_trajectory(drone, step[i]))
+        i += 1
     build_drone_space(ax, trajectories)
 
     ax.legend()
-    plt.show()
-
+    if args.pdf:
+        #For environment without graphic interface (Output is a .pdf file):
+        plt.savefig(re.sub('.json','.pdf',args.input_file))
```

Comment:
It's better to use [`str.replace()`](https://docs.python.org/3/library/stdtypes.html#str.replace) instead of `re`.